server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -42,99 +41,67 @@
  */
 public class ValuesSourceRegistry {
     public static class Builder {
-        private Map<String, List<Map.Entry<Predicate<ValuesSourceType>, AggregatorSupplier>>> aggregatorRegistry = new HashMap<>();
+        private Map<String, List<Map.Entry<ValuesSourceType, AggregatorSupplier>>> aggregatorRegistry = new HashMap<>();
```

Comment:
This doesn't need to be a List now.  We can just make it `Map<String, Map<ValuesSourceType, AggregatorSupplier>>`.  The list was only necessary because lambdas make bad hash keys.

Reply:
Can I do it later in a separate PR? This will really mess me up if it is done in this PR.

Reply:
No problem @imotov if you are both ok with this, I will merge this PR and you can change this List to Map later.